datasets/duconv/duconv.py

```diff
@@ -0,0 +1,134 @@
+# coding=utf-8
+# Copyright 2020 The TensorFlow datasets Authors and the HuggingFace datasets, DataLab Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from email import message
+import json
+import datalabs
+from datalabs import get_task, TaskType
+from datalabs.features import ClassLabel, Features, Sequence, Value
+import os
+logger = datalabs.logging.get_logger(__name__)
+
+_DESCRIPTION = """\
+DuConv is a large-scale proactive human-machine conversation dataset. It contains 270K utterances of 30K dialogues, with one agent proactively leading the conversation based on the knowledge graph.
+"""
+
+_CITATION = """\
+@article{wu2019proactive,
+  title={Proactive human-machine conversation with explicit conversation goals},
+  author={Wu, Wenquan and Guo, Zhen and Zhou, Xiangyang and Wu, Hua and Zhang, Xiyuan and Lian, Rongzhong and Wang, Haifeng},
+  journal={arXiv preprint arXiv:1906.05572},
+  year={2019}
+}
+"""
+
+_LICENSE = "NA"
+
+_TRAIN_DOWNLOAD_URL = "https://cdatalab1.oss-cn-beijing.aliyuncs.com/dialogue/duconv/train.txt"
+_VALIDATION_DOWNLOAD_URL = "https://cdatalab1.oss-cn-beijing.aliyuncs.com/dialogue/duconv/dev.txt"
+_TEST_DOWNLOAD_URL = "https://cdatalab1.oss-cn-beijing.aliyuncs.com/dialogue/duconv/test.txt"
+
+_HOMEPAGE = "https://ai.baidu.com/broad/subordinate?dataset=duconv"
+
+class DuconvConfig(datalabs.BuilderConfig):
+
+    def __init__(self, **kwargs):
+
+        super(DuconvConfig, self).__init__(**kwargs)
+
+class Duconv(datalabs.GeneratorBasedBuilder):
+
+    BUILDER_CONFIGS = [
+        DuconvConfig(
+            name="train",
+            version=datalabs.Version("1.0.0"),
+            description="train set",
+        ),
+        DuconvConfig(
+            name="dev",
+            version=datalabs.Version("1.0.0"),
+            description="dev set",
+        ),
+        DuconvConfig(
+            name="test",
+            version=datalabs.Version("1.0.0"),
+            description="test set",
+            
+        ),
+    ]
+
+    def _info(self):
+        features = {feature: Sequence(Value("string")) for feature in ["goal", "knowledge", "content"]}
```

Comment:
the schema of features should be declared in `TaskType.knowledge_driven_dialogue` template.